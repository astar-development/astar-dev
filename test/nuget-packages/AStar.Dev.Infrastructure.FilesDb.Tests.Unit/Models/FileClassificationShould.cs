using AStar.Dev.Utilities;
using JetBrains.Annotations;

namespace AStar.Dev.Infrastructure.FilesDb.Models;

[TestSubject(typeof(FileClassification))]
public class FileClassificationShould
{
    [Fact]
    public void ContainTheExpectedProperties()
        => new FileClassification
           {
               Id        = 1,
               Celebrity = true,
               Name      = "Mock Name",
               FileDetails   = new List<FileDetail>
                               {
                                   new() { Id    = 1, DirectoryName = "Mock Directory", FileName = "Mock File Name", UpdatedOn = new (new (2025, 6, 28, 22, 20, 37, DateTimeKind.Utc)) }
                               },
               FileNameParts = new List<FileNamePart> { new()  { Id = 1, Text          = "Mock Text Part" } }
           }.ToJson().ShouldMatchApproved();
}
