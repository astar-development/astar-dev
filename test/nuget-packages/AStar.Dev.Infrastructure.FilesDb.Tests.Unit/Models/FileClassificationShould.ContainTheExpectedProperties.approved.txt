{
  "id": 1,
  "name": "Mock Name",
  "fileDetails": [
    {
      "fileClassifications": [],
      "id": 1,
      "imageDetails": {
        "width": null,
        "height": null
      },
      "fileName": "Mock File Name",
      "directoryName": "Mock Directory",
      "fullNameWithPath": "Mock Directory/Mock File Name",
      "fileSize": 0,
      "isImage": false,
      "fileHandle": "",
      "fileCreated": "0001-01-01T00:00:00+00:00",
      "fileLastModified": "0001-01-01T00:00:00+00:00",
      "fileLastViewed": null,
      "softDeleted": false,
      "softDeletePending": false,
      "moveRequired": false,
      "hardDeletePending": false,
      "hardDeleted": false,
      "updatedBy": "Jay Barden",
      "updatedOn": "2025-06-28T22:20:37+00:00"
    }
  ],
  "celebrity": true,
  "fileNameParts": [
    {
      "id": 1,
      "text": "Mock Text Part",
      "fileClassifications": []
    }
  ]
}